src/lib/core/overlay/overlay-directives.ts

```diff
@@ -172,6 +144,20 @@ export class ConnectedOverlayDirective implements OnDestroy {
     this._destroyOverlay();
   }
 
+  ngOnChanges(changes: SimpleChanges) {
+    if (changes['open']) {
+      changes['open'].currentValue ? this._attachOverlay() : this._detachOverlay();
```

Comment:
After you check `changes['open']`, shouldn't you be able to just use `this.open`?
(same for others)